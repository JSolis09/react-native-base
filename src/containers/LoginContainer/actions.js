import {
    LOGIN,
    LOGIN_SUCCESSFUL,
    LOGIN_FAILED
} from './constants';
export const loginAction = (email, password) => ({
    type: LOGIN,
    auth: {
        email,
        password
    }
});

export const loginSuccessfulAction = (user) => ({
    type: LOGIN_SUCCESSFUL,
    user
});

export const loginFailedAction = (message) => ({
    type: LOGIN_FAILED,
    message
});