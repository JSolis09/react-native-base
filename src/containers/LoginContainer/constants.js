export const LOGIN = 'LoginContainer/LOGIN';
export const LOGIN_SUCCESSFUL = 'LoginContainer/LOGIN_SUCCESSFUL';
export const LOGIN_FAILED = 'LoginContainer/LOGIN_FAILED';