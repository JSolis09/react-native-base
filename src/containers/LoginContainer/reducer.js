import {
    LOGIN,
    LOGIN_SUCCESSFUL,
    LOGIN_FAILED
} from './constants';

const initialState = {
    alert: {
        visible: false,
        message: '',
        type: ''
    },
    user: {},
    loading: false
};

export default function LoginContainerReducer(state = initialState, action) {
    switch(action.type) {
        case LOGIN:
            return {
                ...state,
                alert: { visible: false, message: ''},
                loading: true
            };
        case LOGIN_SUCCESSFUL:
            const alertSuccess = {
                visible: true,
                message: 'Login Successful!!',
                type: 'success'
            };
            return {
                ...state,
                alert: alertSuccess,
                loading: false,
                user: action.user
            };
        case LOGIN_FAILED:
            const alertFailed = {
                visible: true,
                message: action.message,
                type: 'error'
            };
            return {
                ...state,
                alert: alertFailed,
                loading: false
            };
        default:
            return state;
    }
}