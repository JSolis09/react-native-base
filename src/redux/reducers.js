import { combineReducers } from "redux";
import HomeContainerReducer from "../containers/HomeContainer/reducer";
import LoginContainerReducer from "../containers/LoginContainer/reducer";

export default combineReducers({
  home: HomeContainerReducer,
  auth: LoginContainerReducer
});
