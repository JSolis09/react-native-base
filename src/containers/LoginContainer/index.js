import React, {Component} from 'react';
import { connect } from "react-redux";

import { loginAction } from './actions';
import Login from '../../components/Login';

class LoginContainer extends Component {
    static navigationOptions = {
        header: null,
    };

    render() {
        return (<Login {...this.props} />);
    }
}

const mapStateToProps = state => ({
    loading: state.auth.loading,
    alert: state.auth.alert
});
export const mapDispatchToProps = dispatch => ({
    login: (email, password) => dispatch(loginAction(email, password))
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginContainer);