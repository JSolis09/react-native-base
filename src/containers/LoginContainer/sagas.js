import { all, call, put, takeEvery } from 'redux-saga/effects';
import { LOGIN, LOGIN_SUCCESSFUL } from './constants';
import { StackActions, NavigationActions } from 'react-navigation';

import { loginSuccessfulAction, loginFailedAction } from './actions';
import firebase from '../../firebase/FirebaseService';
import NavigationService from '../../utils/NavigationService';

export function authentication(email, password) {
    return firebase.auth()
        .signInWithEmailAndPassword(email, password);
}

export function* loginSaga({ auth }) {
    const { email, password } = auth;
    try {
        const user = yield call(authentication, email, password);
        yield put(loginSuccessfulAction(user));
    } catch(e) {
        yield put(loginFailedAction(e.message));
    }
}

export function goHomeSaga() {
    setTimeout(() => {
        const resetAction = StackActions.reset({
            index: 0,
            actions: [NavigationActions.navigate({ routeName: 'Home' })],
        });
        NavigationService.dispatchEvent(resetAction);
    }, 3000);
}

export default function* rootSaga() {
    yield all([
        takeEvery(LOGIN, loginSaga),
        takeEvery(LOGIN_SUCCESSFUL, goHomeSaga)
    ]);
}