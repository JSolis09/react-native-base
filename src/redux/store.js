import { createStore, applyMiddleware } from "redux";
import createSagaMiddleware from 'redux-saga';
import Reducer from "./reducers";

import homeSagas from '../containers/HomeContainer/sagas';
import loginSagas from '../containers/LoginContainer/sagas';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(Reducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(homeSagas);
sagaMiddleware.run(loginSagas);

export default store;