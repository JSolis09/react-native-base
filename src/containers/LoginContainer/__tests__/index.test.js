import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from "redux";

import Reducer from '../../../redux/reducers';
import LoginContainer from '../index';

describe('LoginContainer', () => {
    let store = createStore(Reducer);
    it('should render correctly', () => {
        const tree = create(
            <Provider store={store}>
                <LoginContainer />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});