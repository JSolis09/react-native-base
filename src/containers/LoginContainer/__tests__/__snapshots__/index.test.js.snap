// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "height": 48,
          "justifyContent": "center",
        },
        Object {
          "paddingLeft": 16,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 14,
            "fontWeight": "500",
            "lineHeight": 24,
          },
          Object {
            "color": "#2196f3",
            "fontSize": 20,
          },
        ]
      }
    >
      Log In
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email address"
    style={
      Object {
        "borderBottomColor": "#2196f3",
        "borderBottomWidth": 1,
        "borderStyle": "solid",
        "fontSize": 15,
        "marginBottom": 10,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      Object {
        "borderBottomColor": "#2196f3",
        "borderBottomWidth": 1,
        "borderStyle": "solid",
        "fontSize": 15,
        "marginBottom": 10,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 2,
          "flexDirection": "row",
          "height": 36,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        },
        false,
        Object {
          "backgroundColor": "#fff",
          "borderColor": "rgba(0,0,0,.12)",
        },
        false,
        false,
        Object {
          "backgroundColor": "#2196f3",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
          "zIndex": 1,
        },
        Object {
          "marginTop": 30,
        },
      ]
    }
    testID={null}
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "500",
          },
          false,
          undefined,
          false,
          false,
          Object {
            "color": "#ffffff",
          },
          Object {
            "fontSize": 15,
          },
        ]
      }
    >
      LOG IN
    </Text>
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.87)",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
      <View
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.87)",
            "borderRadius": 100,
            "height": 200,
            "left": -100,
            "opacity": 0.12,
            "position": "absolute",
            "top": -100,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
            "width": 200,
            "zIndex": 1,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": null,
        "bottom": 0,
        "height": 0,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "zIndex": 9999,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": null,
          "bottom": -9999,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "flex": 1,
              "fontSize": 14,
              "paddingBottom": 14,
              "paddingLeft": 20,
              "paddingTop": 14,
            },
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
